src/NuGet.Core/NuGet.Build.Tasks.Console/MSBuildStaticGraphRestore.cs

```diff
@@ -1037,5 +1034,42 @@ private static IEnumerable<IMSBuildItem> GetDistinctItemsOrEmpty(IMSBuildProject
         {
             return project.GetItems(itemName)?.Distinct(ProjectItemInstanceEvaluatedIncludeComparer.Instance) ?? Enumerable.Empty<IMSBuildItem>();
         }
+
+        /// <summary>
+        /// Logs an error from the specified exception.
+        /// </summary>
+        /// <param name="exception">The <see cref="Exception" /> with details to be logged.</param>
+        private void LogErrorFromException(Exception exception)
+        {
+            switch (exception)
+            {
+                case AggregateException aggregateException:
+                    foreach (Exception innerException in aggregateException.InnerExceptions)
+                    {
+                        LogErrorFromException(innerException);
+                    }
```

Comment:
It looks like `AggregateException` instances can be flattened and handled in just one loop using [Flatten](https://learn.microsoft.com/en-us/dotnet/api/system.aggregateexception.flatten?view=net-6.0) method.